Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -1105,11 +1098,8 @@ protected class ClassWriter : IClassWriter {
           break;
         case SpecialField.ID.ArrayLength:
         case SpecialField.ID.ArrayLengthInt:
-          preString = "len(";
-          postString = ")";
-          if (idParam != null && (int)idParam > 0) {
-            postString = string.Concat(Enumerable.Repeat("[0]", (int)idParam)) + postString;
-          }
+          var dim = idParam is int d and > 0 ? d : 0;
+          compiledName = $"length({dim})";
```

Comment:
Shouldn't this be `self.length`?
```suggestion
          compiledName = $"self.length({dim})";
```